components/nexus-core/src/main/java/org/sonatype/nexus/proxy/maven/maven2/M2Repository.java

```diff
@@ -159,7 +159,8 @@ public boolean isMavenArtifactChecksumPath(String path) {
   @Override
   public boolean shouldServeByPolicies(ResourceStoreRequest request) {
     if (M2ArtifactRecognizer.isMetadata(request.getRequestPath())) {
-      if (M2ArtifactRecognizer.isSnapshot(request.getRequestPath())) {
+      Gav gav = getGavCalculator().pathToGav(request.getRequestPath());
```

Comment:
Because metadata must go in it's proper place too (rel or snap). The problem with metadata, that unlike to artifact paths, the are "incomplete", and "reverse engineer" GAV from them is impossible.